src/rust/terminusdb-community/src/graphql/frame.rs

```diff
@@ -989,4 +1058,43 @@ json{ '@context':_{ '@base':"terminusdb://system/data/",
         // at least it parses!
         // TODO test something here
     }
+
+    #[test]
+    fn invert_frames() {
+        let engine = Engine::new();
+        let activation = engine.activate();
+        let context: Context<_> = activation.into();
+
+        let term = r#"json{
+                   '@context': json{'@base':"terminusdb://woql/data/",'@schema':"http://terminusdb.com/schema/woql#",'@type':'Context'},
+                   'Bar' : json{ '@type' : "Class", elt : "xsd:string" },
+                   'Foo' : json{ '@type' : "Class", a : "Bar", b: json{ '@type' : "Optional", '@class' : "Bar"}, c: json{ '@type' : "Set", '@class' : "Bar"}}}"#;
+        let term = unwrap_result(&context, context.term_from_string(term));
+        let mut allframes: AllFrames = context.deserialize_from_term(&term).unwrap();
+        allframes.invert();
+        assert_eq!(
+            allframes.inverted.as_ref().unwrap().classes["Bar"].domain["a"].class,
```

Comment:
this is backwards, asserts should start with the expected value, then have the actual value